### PR TITLE
Qol/small qol fixes

### PR DESCRIPTION
**Pull request by JstDust**
A bunch of fixes, these were done because i got some user feedback after some use, some are QOL some are UI bug fixes.

- UI BUGs: see "fix search history" & "bug squashing" commits.
- Altered client schema loading to sort schema and tables on name (case insensitive), note: not columns because order can be relevant there
- Fixed issue that all was expanded by default by making a explicit true check on the 'expanded[row.id]', note: it was not consistently always expended or collapsed not sure how that can happen maybe some object pollution with something else then true | false | undefined? (maybe this object should just be a 'Set<string>' instead)
- Altered the schema search to actually search schemas as well if matched tables search is skipped on that schema. and removed schemas that don't yield tables from the table search this helps with dbs with lots of schemas.
- Compression added to the schema's route, this was done because i had a schemas request that was 30-40 mb response, adding the compression resulted in a transfer of only 3.5 mb see commit 'Reduce schema load bottleneck.'.

### CODE REVIEW

**Review by dengc367 — DISMISSED** [review]

**Comment by rickbergfalk** [comment]
Thanks for this @JstDust and for reviewing @dengc367!

It seems tests are failing because of package-lock changes. I have been holding off using npm 7 but I suppose its inevitable and I should embrace it. Will see if I can get tests passing before merge.

File: server/package-lock.json
Comment (rickbergfalk) [inline server/package-lock.json:4]: @JstDust Would it be possible to revert this lockfileVersion 2 change and keep it lockfileVersion 1? This might require installing npm 6 temporarily. If this is problematic I can pull into another branch, make changes, then merge.

File: server/routes/connection-schema.js
Comment (rickbergfalk) [inline server/routes/connection-schema.js:53]: While this would be better for smaller API responses, this would not be compatible with autocomplete unfortunately, as autocomplete requires knowledge of the full schema today. Could you either remove this portion of the comment or add a clarification about autocomplete?

**Review by rickbergfalk — CHANGES_REQUESTED** [review]

File: server/routes/connection-schema.js
Comment (JstDust) [reply server/routes/connection-schema.js]: Hi for now i have removed the comment from the code since adding it there without proper design was not a good plan, sorry for the confusion. 

My though behind that was not alter the clients ability to load the schema. Only the way it is retrieved, currently it is an all or nothing approach this limits the users feedback on what is happening. I do not mind that all is loaded but it should be done with a lot more user feedback. (e.g. a proper loader)
- 'Retrieving schema names....'
- 'Retrieving tables/columns for schema x/x columns retrieved x' << update as data loading promises are completed.
- Show tree

An other advantage of this staged approach is timeout of the client request is less likely to happen due to long wait or abort due to impatient user refreshing the page.

Also the current setup caches the schema flat this is something that could be more structured. Instead of a plain cache it should be considered to load the entire schema into the backend database of sqlpad. Since each driver could provide a different way of loading the data (and be potentially very time consuming) but every driver always presents it the same way. This would allow for searching and loading of schemas to happen against the backend-database that can be indexed. e.g. a tables structure like:
```
connection_schema_version {
 id,
 connection_id,
 timestamp
}
connection_schema (
 connection_schema_version_id,
 schema_name,
 table_name,
 column_name,
 column_type
)
```
The forced refresh can then load a new version and once completed can delete the old version.

File: server/package-lock.json
Comment (JstDust) [reply server/package-lock.json]: @rickbergfalk done

**Review by rickbergfalk — APPROVED** [review]
Thanks for the changes @JstDust!

**Pull request merged**
